stix2validator/shoulds.py

```diff
@@ -274,14 +274,16 @@ def relationships_strict(instance):
         return
 
     r_type = instance['relationship_type']
-    r_source = re.search("(.+)\-\-", instance['source_ref']).group(1)
-    r_target = re.search("(.+)\-\-", instance['target_ref']).group(1)
+    try:
+        r_source = re.search("(.+)\-\-", instance['source_ref']).group(1)
+        r_target = re.search("(.+)\-\-", instance['target_ref']).group(1)
+    except (AttributeError, TypeError):
```

Comment:
Why do we return early from this function? What is "these" that the schemas catch? Is it that if the source or target identifier is bad, we know it won't be an spec-defined relationship?